### PR TITLE
Additions to dependency locking documentation

### PR DESCRIPTION
**Pull request by ljacomet**

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/dependencyLocking.adoc [hunk 1]
```diff
@@ -51,14 +61,14 @@ Or the following, as a way to lock all configurations:
 
 [NOTE]
 ====
-Only configurations that can be resolved will have a lock file attached to them. Applying locking on non resolvable-configurations is simply a no-op.
+Only configurations that can be resolved will have lock state attached to them. Applying locking on non resolvable-configurations is simply a no-op.
 ====
 
-=== Generating and updating lockfiles
+=== Generating and updating lock state
```
Comment (melix) [inline subprojects/docs/src/docs/userguide/dependencyLocking.adoc]: Maybe "dependency locks" would sound better from an external point of view. No strong opinion.

**Comment by ljacomet** [comment]
@eriwen Because what matters is the lock state, not the fact that it is represented by a file for now.

File: subprojects/docs/src/docs/userguide/dependencyLocking.adoc (hunk 1 above)
Comment (ljacomet) [reply subprojects/docs/src/docs/userguide/dependencyLocking.adoc]: Good point, changing

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -46,9 +46,17 @@ Add-->
 ### Locking of dynamic dependencies
 
 Gradle now provides a mechanism for locking dynamic versions.
-It enables builds to become reproducible even when declaring dependencies using version ranges.
+It makes builds reproducible even when declaring dependencies using version ranges.
 
-You first enable locking on configurations:
+This enables, amongst others, the following scenarios:
+
+* Companies dealing with multi repositories no longer need to rely on `-SNAPSHOT` or changing dependencies,
```
Comment (pioterj) [inline subprojects/docs/src/docs/release/notes.md]: Didn't you mean "for" instead of "or" before "changing dependencies"?

**Comment by pioterj** [comment]
@ljacomet Much better, thanks. I left some comments.

File: subprojects/docs/src/docs/release/notes.md (hunk 2 above)
Comment (ljacomet) [reply subprojects/docs/src/docs/release/notes.md]: No, while `SNAPSHOT` are implicitly changing, you could have a company using its own scheme, non maven, for having changing dependencies.

**Pull request merged**